### PR TITLE
Selected exercises

### PR DESCRIPTION
**Pull request by xus84**
I added the functionality to selected exercises, it shows a border and the exercise appears in a list.

### CODE REVIEW

File: src/app/exercises/ExerciseCard.js [hunk 1]
```diff
@@ -1,16 +1,35 @@
 "use client"
-const ExerciseCard = ({ exercise}) => {
+import { useState } from 'react';
+const ExerciseCard = ({ exercise, onSelect }) => {
```
Comment (smarcilla) [inline src/app/exercises/ExerciseCard.js]: Certainly! Here are some specific recommendations for the `ExerciseCard` component in English, which you can consider for improving the code:

1. **Consolidate Styling Approaches**:
   - Mix of inline styles and Tailwind CSS classes can be confusing. Consider using Tailwind classes consistently for better maintainability and readability.
   - Example: Replace inline styles for `border`, `padding`, `margin`, etc., with Tailwind utility classes.

2. **Refactor Conditional Inline Styles**:
   - The conditional styling for `border` can be improved. You could create custom Tailwind classes or use CSS for conditionally applied styles.
   - Example: Instead of inline styles, use `className` with template literals to toggle Tailwind classes.

3. **Image Styling**:
   - The inline styles for image dimensions (`{ width: exercise.width, height: exercise.height }`) might be better handled in CSS if these values are static. If they are dynamic, ensure they are correctly applied.
   - Example: If `exercise.width` and `exercise.height` are static, define them in a CSS class.

4. **Improve Accessibility**:
   - Ensure the `alt` text for images is descriptive for better accessibility.
   - Example: Check if `alt={name}` is descriptive enough or if a more detailed description is needed.

5. **Review State Management**:
   - Ensure the `isClicked` state and `onSelect` callback are necessary and properly handled. If `isClicked` is only relevant to this component, consider whether it’s essential.
   - Example: Re-evaluate if the `isClicked` state can be managed more efficiently or if it's redundant.

6. **Component Documentation**:
   - Commenting on the purpose of the component and its props (`exercise`, `onSelect`) can be very helpful for other developers.
   - Example: Add a comment block at the beginning of the component explaining its usage and props.

7. **Responsive Design Check**:
   - Verify that the component layout is responsive and appears correctly on various screen sizes.
   - Example: Test the component in different viewports to ensure it's visually appealing and functional.

8. **Code Comments for Suggestions**:
   - Place comments near the specific lines or blocks where changes are suggested for clarity.
   - Example: Add a comment above the image element suggesting the use of conditional Tailwind classes for border styling.

These suggestions aim to improve the overall quality, maintainability, and readability of your `ExerciseCard` component. Remember, the best practices depend on your project's specific needs and the conventions your team follows.

File: src/app/exercises/page.js [hunk 2]
```diff
@@ -1,84 +1,139 @@
-import React from 'react'
+"use client"
+import React, { useEffect, useState } from 'react'
 
 import ExerciseCard from './ExerciseCard';
 import Header from '@/components/Header';
 
 function Exercises() {
```
Comment (smarcilla) [inline src/app/exercises/page.js:6]: ## Code Review and Suggestions

### 1. Key Prop in Map Function
- It's recommended to avoid using the index as a `key` in the `.map()` function. Consider using unique identifiers like `exercise.id` for keys to prevent potential issues with rendering and state management.

### 2. Handling Development Logs
- The `useEffect` hook used for logging `selectedExercises` is helpful during development but should be removed or commented out in production code for cleaner and more efficient code.

### 3. Static Exercise Data
- The exercise data is hard-coded within the component. For scalability and maintainability, consider fetching this data from an external source or moving it to a separate file or a state management solution.

### 4. Code Consistency and Styling
- Ensure that the styling and class naming conventions are consistent throughout the project. Tailwind CSS is being used effectively, but always double-check for consistency.

### 5. Accessibility and Semantic HTML
- Review the component for accessibility considerations. Use semantic HTML where possible and include `aria` attributes for enhanced accessibility, ensuring the application is usable by all users.

Implementing these suggestions will improve the overall quality, maintainability, and accessibility of the `Exercises` component.

File: src/app/layout.js [hunk 3]
```diff
@@ -1,8 +1,8 @@
-import { Inter } from 'next/font/google'
+import { Lato } from 'next/font/google'
 import Head from 'next/head';
 import './globals.css'
 
-const inter = Inter({ subsets: ['latin'] })
+const lato = Lato({ subsets: [], weight: ['400', '700'] })
 
 export const metadata = {
```
Comment (smarcilla) [inline src/app/layout.js:7]: ## Code Review and Suggestions

### 1. Next.js Head Component Usage
- Good use of the Next.js `Head` component for including the favicon link. Consider also using it to include other metadata from the `metadata` object, like the title and description for SEO purposes.

### 2. Font Import and Usage
- Ensure that the syntax for importing and using the Lato font is up to date with the latest Next.js practices. Verify that the font is loading correctly on all pages.

### 3. Document Structure and Metadata
- The `<html>` tag is generally not used directly in React components as Next.js handles this. Double-check if this is necessary in your case.
- Utilize the `metadata` object effectively by integrating it into the `Head` component for dynamic metadata rendering.

### 4. Global Styles and Consistency
- Confirm that applying the Lato font class to the entire `<body>` is intentional and that it's being applied as expected.
- Review the global styles import to ensure that all global styles are correctly applied across the application.

### 5. Accessibility and Semantic HTML
- Consider adding more SEO and accessibility-friendly metadata in the `Head`, such as meta descriptions and other relevant tags to enhance the user experience and site visibility.

Implementing these suggestions will enhance the functionality, accessibility, and SEO friendliness of your application.

**Review by smarcilla — CHANGES_REQUESTED** [review]
## Overall Review

This review covers several key aspects of the code that could be improved for better performance, maintainability, and adherence to best practices. 

### Recommendations:

1. **Font Import and Usage**: Verify that the method of importing and using the Google font is current with Next.js best practices. Ensure that the font loads correctly across the application.

2. **Use of `<html>` Tag**: In Next.js, the `<html>` tag is typically managed by the framework itself. Please confirm if there's a specific reason for including it directly in the `RootLayout` component.

3. **Metadata Implementation**: The `metadata` object contains important information that is not being used within the `<Head>` tag. Consider using these properties to dynamically set page metadata for SEO.

4. **Global Styles**: Ensure that the global styles and font classes are applied correctly and consistently. The `globals.css` should handle all the universal styling needs across the app.

5. **Accessibility**: Adding more descriptive metadata and ensuring semantic HTML structure will enhance the accessibility of the site.

It's great to see such foundational work being put into the application's structure. Implementing the above suggestions will not only align the code with the latest practices but also improve the scalability and user experience of the application.

**Comment by xus84** [comment]
Nice, I did some changes, you can check them. ✔️

File: src/app/exercises/ExerciseCard.js (hunk 1 above)
Comment (smarcilla) [reply src/app/exercises/ExerciseCard.js]: ### What You've Done Well:

1. **State Management with `useState`**:
   - You're using something called `useState` to keep track of whether a card has been clicked or not. It's like having a small memory inside your component that remembers if it's been clicked.

2. **Structure and Use of Data (Props)**:
   - Your component is well-organized and uses something called `props` (like `exercise` and `onSelect`). It's like receiving gifts (data) that you can use in your component.

3. **Responding to Clicks**:
   - You make something happen when someone clicks on a card. This is great because it makes your component interactive.

### Areas to Improve:

1. **Consistent Styling**:
   - Right now, you're mixing two ways of adding styles: directly in your code and using Tailwind CSS (a styling tool). It's better to choose one to make everything clearer and easier to maintain.

2. **Conditionally Changing Styles**:
   - When the 'clicked' state changes, you also change the style of the border. But you're doing this directly in your code. You could use Tailwind to make these changes, which would clean up your code.

3. **Image Sizing**:
   - If your image sizes are always the same, you can put these styles in a separate CSS file. If they change, it's okay to leave it as is, but make sure it works well.

4. **Making Your Site Friendly for Everyone**:
   - Ensure that the descriptions of your images (the `alt` text) are helpful for people who can't see the images. This makes your site more friendly for everyone.

5. **Explain Your Code**:
   - It would be great if you add comments in your code explaining what your component does and what those `props` are. It's like leaving notes for yourself or for others who might see your code in the future.

6. **Design That Works on All Screens**:
   - Check that your component looks good on different screen sizes, like mobile phones and tablets. This is important so everyone can use your site without problems.

In summary, you're doing a good job with the basic parts of your component. You just need some improvements in how you handle styles and in making your code clearer and more accessible for everyone. Keep it up, and your component will be even better!

File: src/app/exercises/page.js (hunk 2 above)
Comment (smarcilla) [reply src/app/exercises/page.js:6]: ### Corrected Points:

1. **Unique Keys for Mapping**: 
   - You're using `exercise.id` as the key in your `map` function, which is good! This is better than using indexes because it helps React know each item uniquely, especially when updating or deleting items.

2. **Separation of Components**: 
   - You've done a great job separating your components like `ExerciseCard` and `Header`. This makes your code organized and easier to manage.

3. **Interactive Functionality**: 
   - Your `handleExerciseClick` function works well for selecting and deselecting exercises. It's a smart way to toggle items in a list, which is a common task in web apps.

### Points Not Addressed:

1. **Removing Development Logs**:
   - The `useEffect` for logging selected exercises is useful for development, but remember to remove or comment it out for the final version. It's like cleaning up your workspace after you're done working.

2. **External Data Source**:
   - Your exercise data is still directly in your component. In the future, consider getting this data from an external source, like a file or a database. It's like keeping your music on a playlist online instead of having it all on your phone.

3. **Accessibility and HTML Structure**:
   - Check if your HTML is easy to understand and accessible. Use proper HTML elements and add attributes that help people with disabilities. Think of it as making sure everyone can read and understand a public sign.

### Simplified and Translated:

You're doing well with using unique IDs in your lists and separating your code into different parts. It's like organizing your music by playlists. The click function for choosing exercises is cool and handy, like having a remote to switch channels on your TV.

Remember to clean up your code by removing extra logs before sharing your final project, like tidying up your room when expecting guests. Consider storing your exercise data somewhere else, like having your music on a streaming service instead of on your phone. And make sure your website is easy for everyone to use, just like ensuring everyone can read and understand signs in a public place.

Comment (xus84) [reply src/app/exercises/page.js:6]: I reviewed the points not addressed: 

1. The useEffect was removed.
2. External Data source, I think that at the moment using a file with the exercises is enough.
3. HTML issue, I can't change this part because nextjs 14 version use its own way to manage metadata. 

To me I think it is possible to merge it.

**Review by smarcilla — APPROVED** [review]
